docs/docs/integrations/chat/openai.ipynb

```diff
@@ -915,6 +915,174 @@
     "response_2.text()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "34ad0015-688c-4274-be55-93268b44f558",
+   "metadata": {},
+   "source": [
+    "#### Code interpreter\n",
+    "\n",
+    "OpenAI implements a [code interpreter](https://platform.openai.com/docs/guides/tools-code-interpreter) tool to support the sandboxed generation and execution of code.\n",
+    "\n",
+    "Example use:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "34826aae-6d48-4b84-bc00-89594a87d461",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_openai import ChatOpenAI\n",
+    "\n",
+    "llm = ChatOpenAI(model=\"o4-mini\", use_responses_api=True)\n",
+    "\n",
+    "llm_with_tools = llm.bind_tools(\n",
+    "    [\n",
+    "        {\n",
+    "            \"type\": \"code_interpreter\",\n",
+    "            # Create a new container\n",
+    "            \"container\": {\"type\": \"auto\"},\n",
+    "        }\n",
+    "    ]\n",
+    ")\n",
+    "response = llm_with_tools.invoke(\n",
+    "    \"Write and run code to answer the question: what is 3^3?\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1b4d92b9-941f-4d54-93a5-b0c73afd66b2",
+   "metadata": {},
+   "source": [
+    "Note that the above command created a new container. We can also specify an existing container ID:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "d8c82895-5011-4062-a1bb-278ec91321e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tool_outputs = response.additional_kwargs[\"tool_outputs\"]\n",
+    "assert len(tool_outputs) == 1\n",
+    "container_id = tool_outputs[0][\"container_id\"]\n",
```

Comment:
Maybe highlight this one as well? I might be just tired, but I didn't notice this line when I was skimming through the code snippet looking directly at the highlighted line below